packages/teleport/src/Discover/Kubernetes/HelmChart/HelmChart.story.tsx

```diff
@@ -0,0 +1,154 @@
+/**
+ * Copyright 2022 Gravitational, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from 'react';
+import { MemoryRouter } from 'react-router';
+
+import { ContextProvider, Context as TeleportContext } from 'teleport';
+import cfg from 'teleport/config';
+import { ResourceKind } from 'teleport/Discover/Shared';
+import {
+  clearCachedJoinTokenResult,
+  JoinTokenProvider,
+} from 'teleport/Discover/Shared/JoinTokenContext';
+import { PingTeleportProvider } from 'teleport/Discover/Shared/PingTeleportContext';
+import { userContext } from 'teleport/mocks/contexts';
+
+import HelmChart from './HelmChart';
+
+export default {
+  title: 'Teleport/Discover/Kube/HelmChart',
+  decorators: [
+    Story => {
+      // Reset request handlers added in individual stories.
+      window.msw.worker.resetHandlers();
+      clearCachedJoinTokenResult();
+      return <Story />;
+    },
+  ],
+};
+
+export const Init = () => {
+  return (
+    <Provider>
+      <HelmChart />
+    </Provider>
+  );
+};
+
+export const Polling = () => {
+  const { worker, rest } = window.msw;
+  // Use default fetch token handler defined in mocks/handlers
+
+  worker.use(
+    rest.get(cfg.api.kubernetesPath, (req, res, ctx) => {
+      return res(ctx.delay('infinite'));
+    })
+  );
+  return (
+    <Provider>
+      <HelmChart runJoinTokenPromise={true} />
+    </Provider>
+  );
+};
+
+export const PollingSuccess = () => {
+  const { worker, rest } = window.msw;
+  // Use default fetch token handler defined in mocks/handlers
+
+  worker.use(
+    rest.get(cfg.api.kubernetesPath, (req, res, ctx) => {
+      return res(ctx.json({ items: [{}] }));
+    })
+  );
+  return (
+    <Provider interval={5}>
+      <HelmChart runJoinTokenPromise={true} />
+    </Provider>
+  );
+};
+
+export const PollingError = () => {
+  const { worker, rest } = window.msw;
+  // Use default fetch token handler defined in mocks/handlers
+
+  worker.use(
+    rest.get(cfg.api.kubernetesPath, (req, res, ctx) => {
+      return res(ctx.delay('infinite'));
+    })
+  );
+  return (
+    <Provider timeout={20}>
+      <HelmChart runJoinTokenPromise={true} />
+    </Provider>
+  );
+};
+export const Processing = () => {
+  const { worker, rest } = window.msw;
```

Comment:
[nit]
```suggestion
const { worker, rest } = window.msw;
export const Processing = () => {
```
can be moved at module level to avoid repetition in other stories. 

Whats the reason behind assigning `msw` to `window`? I think it could work nicely as module:
```
import { worker, rest } 'utils/msw'
```

Reply:
i guess i didn't want the worker to be specific to teleport (like if teleterm wanted to define their handlers, though not sure how that works with grpc) 

.storybook/preview.js
```
// Checks we are running non-node environment (browser)
if (typeof global.process === 'undefined') {
  const worker = setupWorker(...handlersTeleport);  <--- can join more handlers
  worker.start();

  // So it can be accessed in stories more easily.
  window.msw = { worker, rest };
}
```